pyroclast/common/visualizable.py

```diff
@@ -0,0 +1,94 @@
+import abc
+import numpy as np
+import tensorflow as tf
+
+
+class VisualizableMixin(abc.ABC):
+    """Interface Model interface.
+
+    Provides functionality for various calculations and
+    visualizations. Should be used as a mixin in conjunction with
+    tf.Module.
+
+    """
+
+    @abc.abstractmethod
+    def classify(self, x):
+        """Calculates output logits given a set of inputs.
+
+        Args:
+           x (tf.Tensor): Input of shape [batch_size, ...data_shape]
+
+        Returns:
+            logits (tf.Tensor): shape [batch_size, num_classes]
+        """
+        pass
+
+    @abc.abstractmethod
+    def conv_stack_submodel(self):
+        """Returns the submodel that contains the convolution stack
```

Comment:
Does it return an object which contains the conv stack, or does it return the conv stack?

Reply:
I think that it should contain the final layer in the conv stack. What do you think?